code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -56,8 +61,48 @@
 						return
 					M.put_in_l_hand(src)
 			add_fingerprint(usr)
-			return
 
+//Proc for Contents transfer to other containers, bins, turf, and more possible additions.
+/obj/item/weapon/storage/proc/contentto(obj/dest_object, mob/user)
+	playsound(loc, "rustle", 50, 1, -5)
+	switch( storage_contents_dump_act(dest_object) )
+		if(1) // turf
+			var/turf/T = get_turf(dest_object)
+			for(var/obj/item/I in src)
+				remove_from_storage(I, T)
+			return 1
+
+		if(2) //storage item
+			var/obj/item/weapon/storage/S = dest_object
+			for(var/obj/item/I in src)
+				if(S.can_be_inserted(I,0,user))
+					remove_from_storage(I, S)
+			orient2hud(user)
+			S.orient2hud(user)
+			if(user.s_active) //refresh the HUD to show the transfered contents
+				user.s_active.close(user)
+				user.s_active.show_to(user)
+			return 2
+
+		if(3) //bin
+			for(var/obj/item/I in src)
+				remove_from_storage(I, dest_object) //No check needed, put everything inside
+			return 3
+
+	return 0
+
+//Check the destination item type for contentto.
+/obj/item/weapon/storage/proc/storage_contents_dump_act(obj/dest_object)

```

Comment:
As part of the above change, this would need to be defined as:
```
/atom/proc/storage_contents_dump_act(var/obj/item/weapon/storage/S)
    return 0
````